docs/dev/guidelines/server.rst

```diff
@@ -350,3 +350,240 @@ Please read `the AssertJ documentation <https://assertj.github.io/doc/#assertj-c
+
+General Testing best practices are found on https://phauer.com/2019/modern-best-practices-testing-java/
+
+Also check out https://www.baeldung.com/spring-tests
+
+And https://howtodoinjava.com/best-practices/unit-testing-best-practices-junit-reference-guide/
+
+
+23. Avoid using @MockBean
+=========================
+
+Do not use the ``@SpyBean`` or ``@MockBean`` annotation unless absolutely necessary, or possibly in an abstract Superclass. If you want to see more in detail why, take a look `here. <https://www.baeldung.com/spring-tests>`_
+Basically every time @MockBean appears in a class, the ApplicationContext cache gets marked as dirty, hence the runner will clean the cache after the test-class is done and restarts the application context.
+This leads to a large overhead, which makes the tests tend to take a lot more time.
+
+Here is an example how to replace a ``@SpyBean``. We wanted to test a catch case which is only executed if an IOException gets thrown. We did this by mocking the service method and making it throw an Exception.
+
+.. code-block:: java
+
+    public class TestExport extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+        @SpyBean
+        private FileUploadSubmissionExportService fileUploadSubmissionExportService;
+
+        @Test
+        @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+        public void testExportAll_IOException() throws Exception {
+            doThrow(IOException.class).when(fileUploadSubmissionExportService).export(any(), any());
+            request.postWithResponseBodyFile("/api/file-upload-export/" + fileUploadExercise.getId()", HttpStatus.BAD_REQUEST);
+        }
+    }
+
+This, as mentioned above, we should really avoid this.
+Instead we can use Static Mocks. When we look deeper in the ``export()`` method we find that there is a call of ``File.newOutputStream(..)``.
+Now, instead of mocking the whole Service we can just mock this static method, like this:
+
+.. code-block:: java
+
+    public class TestExport extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+        // No beans used anymore
+        @Test
+        @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+        public void testExportAll_IOException() throws Exception {
+            MockedStatic<Files> mockedFiles = mockStatic(Files.class);
+            mockedFiles.when(() -> Files.newOutputStream(any(), any())).thenThrow(IOException.class);
+            request.postWithResponseBodyFile("/api/file-upload-export/" + fileUploadExercise.getId()", HttpStatus.BAD_REQUEST);
+
+            mockedFiles.close();
+        }
+    }
+
+What you notice here is that we can avoid the use of a Bean and also test deeper. Important to mention is that you need to close the mock at the end of the test again.
+
+For a real examples where a SpyBean was replace with a static mock look `here. <https://github.com/ls1intum/Artemis/commit/4843137aa01cfdf27ea019400c48df00df36ed45>`_
```

Comment:
You could unify to ``@SpyBean` and capitalize Static Mock as above.

Reply:
This is not really different to the in-doc example, is it necessary? If you want to keep it, I would add the file name (an even lines?) to search for the replacement example. So readers do not have to browse through all the changed files. I think you cannot link the exact lines unfortunately.

Reply:
typo: replace_d_

Reply:
```suggestion
For a real example where a SpyBean was replaced with a static mock look at the SubmissionExportIntegrationTest.java in `here. <https://github.com/ls1intum/Artemis/commit/4843137aa01cfdf27ea019400c48df00df36ed45>`_
```

Reply:
good catch :+1: